opm/grid/cpgrid/CpGridUtilities.cpp

```diff
@@ -65,4 +69,100 @@ lgrIJK(const Dune::CpGrid& grid, const std::string& lgr_name)
+
+            // Get min/max k for pillar at (i,j)
+            auto [bottom_k, top_k] = std::minmax_element(pillar.begin(), pillar.end());
+
+            const auto bottom_lgr_cartesian_idx = ((*bottom_k)*nx*ny) + (j*nx) + i;
+            const auto top_lgr_cartesian_idx = ((*top_k)*nx*ny) + (j*nx) + i;
+
+            const auto& bottomElemIdx = lgrCartesianIdxToCellIdx.at(bottom_lgr_cartesian_idx);
+            const auto& topElemIdx = lgrCartesianIdxToCellIdx.at(top_lgr_cartesian_idx);
+
+            const auto& levelGrid = *(grid.currentData()[level]);
+
+            const auto& bottomElem = Dune::cpgrid::Entity<0>(levelGrid, bottomElemIdx, true);
+            const auto& topElem = Dune::cpgrid::Entity<0>(levelGrid, topElemIdx, true);
+
+            // Recall that a cell has 8 corners:
+            //        6 --- 7
+            //       /     /   TOP FACE
+            //      4 --- 5
+            //        2 --- 3
+            //       /     /   BOTTOM FACE
+            //      0 --- 1
+
+            // To take into account inactive cells, consider for each (i,j) group of cells, 4 pillars:
+            // (i,j)     pillar associated with bottom element corner 0 and top element corner 4
+            // (i+1,j)   pillar associated with bottom element corner 1 and top element corner 5
+            // (i,j+1)   pillar associated with bottom element corner 2 and top element corner 6
+            // (i+1,j+1) pillar associated with bottom element corner 3 and top element corner 7
+            Opm::processPillars(i,j, nx, topElem, bottomElem, lgrCOORD);
+        }
+    }
+    return lgrCOORD;
+}
+
+void setPillarCoordinates(int i, int j, int nx,
+                          int topCorner, int bottomCorner, int positionIdx,
+                          const Dune::cpgrid::Entity<0>& topElem,
+                          const Dune::cpgrid::Entity<0>& bottomElem,
+                          std::vector<std::array<double, 6>>& lgrCOORD)
+{
+    const int pillar = ((j + positionIdx / 2) * (nx + 1)) + (i + positionIdx % 2);
+
+    // Top pillar's COORD values
+    const auto& top_point = topElem.subEntity<3>(topCorner).geometry().center();
+    std::copy(top_point.begin(), top_point.end(), lgrCOORD[pillar].begin());
+
+    // Bottom pillar's COORD values
+    const auto& bottom_point = bottomElem.subEntity<3>(bottomCorner).geometry().center();
+    std::copy(bottom_point.begin(), bottom_point.end(), lgrCOORD[pillar].begin() + 3);
+}
+
+void processPillars(int i, int j, int nx,
+                    const Dune::cpgrid::Entity<0>& topElem,
+                    const Dune::cpgrid::Entity<0>& bottomElem,
+                    std::vector<std::array<double, 6>>& lgrCOORD)
+{
+    setPillarCoordinates(i, j, nx, 4, 0,  0, topElem, bottomElem, lgrCOORD);
+    setPillarCoordinates(i, j, nx, 5, 1,  1, topElem, bottomElem, lgrCOORD);
+    setPillarCoordinates(i, j, nx, 6, 2,  2, topElem, bottomElem, lgrCOORD);
```

Comment:
Please add a comment why we need a few of these twice

Reply:
Added!